### ISSUE TITLE
Recursively expand empty dirs when filesystem group_empty_dirs is true

### ISSUE THREAD

**Issue by kittydoor** [nvim-neo-tree/neo-tree.nvim#552]
The initial feature was implemented in #216 and only has one feature missing before it is at feature-parity with other plugins who do something similar, which is not having to <CR> multiple times to open nested folders.

Ideally a single <CR> would initiate a recursive expansion of the dir until a non-empty dir is found.

I understand this is not a priority, just wanted to bring some eyes to this in case someone (maybe even myself sometime in the future) wants to get to it!

**Comment by jayp0521** [comment]
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/35e9f9b53d05bfec90314865787cd920a8b468aa/lua/neo-tree/ui/renderer.lua#L980-L998

I'm new to the internals of this plugin, but based on the above, it seems like this should be happening already. Maybe there is a bug somewhere

**Comment by cseickel** [comment]
> I'm new to the internals of this plugin, but based on the above, it seems like this should be happening already. Maybe there is a bug somewhere

No, that groups empty dirs that were already loaded, but it won't recursively go back out and scan the filesystem until it finds a file.

I did try to enable that functionality originally but decided to skip because it was really difficult to implement with the single level lazy loading strategy that is being used.

**Comment by nyngwang** [comment]
> [...] it was really difficult to implement with the single level lazy loading strategy that is being used.

I think lazy-loading is more important in this case.

I haven't tried it but how about this general idea: neo-tree might have a way to determine whether the cursor node is an un-expanded folder, and you can use a command to move the cursor with `j` after expanding it. This way you just need to press the same key-bind multiple times to expand into the deepest level without the need for `j`s.
